### PR TITLE
hardening/1172_add_ngsihdfssink_configuration_in_docker

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1172 
* Installation process:
```
Successfully built 9af07c91e2d4

REPOSITORY    TAG         IMAGE ID            CREATED             SIZE
cygnus-ngsi   latest      9af07c91e2d4        38 minutes ago      514 MB
centos        6           f07f6ca555a5        6 weeks ago         194.6 MB

$ sudo docker run cygnus-ngsi
....
Component type: CHANNEL, name: mongo-channel started
Component type: CHANNEL, name: sth-channel started
Component type: CHANNEL, name: ckan-channel started
Component type: CHANNEL, name: mysql-channel started
Component type: CHANNEL, name: hdfs-channel started
...

$ docker run -e CYGNUS_LOG_LEVEL='DEBUG' -e CYGNUS_HDFS_HOST='storage.cosmos.lab.fiware.org' -e CYGNUS_HDFS_PORT='14000' -e CYGNUS_HDFS_USER='pcoello25' -e CYGNUS_HDFS_TOKEN='xxxxxxxxxxxxxxxxxxx' cygnus-ngsi
...
[hdfs-sink] Processing sub-batch regarding the pc25serv_/pc25spath_Room1_Room destination
[hdfs-sink] Processing context element (id=Room1, type=Room)
[hdfs-sink] Processing context attribute (name=temperature, type=centigrade)
[hdfs-sink] Persisting data at NGSIHDFSSink. HDFS file (pc25serv/pc25spath/Room1_Room/Room1_Room.txt), Data ({"recvTimeTs":"1477042515","recvTime":"2016-10-21T09:35:15.816Z","fiwareServicePath":"/pc25spath","entityId":"Room1","entityType":"Room","attrName":"temperature","attrType":"centigrade","attrValue":"26.5","attrMd":[]})
Http request: GET http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv/pc25spath/Room1_Room/Room1_Room.txt?op=getfilestatus&user.name=pcoello25 HTTP/1.1
Http response status line: HTTP/1.1 404 Not Found
Http response payload: {"RemoteException""message":"File does not exist: \/user\/pcoello25\/pc25serv\/pc25spath\/Room1_Room\/Room1_Room.txt","exception":"FileNotFoundException","javaClassName":"java.io.FileNotFoundException"}}
Http request: PUT http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv/pc25spath/Room1_Room?op=mkdirs&user.name=pcoello25 HTTP/1.1
Http response status line: HTTP/1.1 200 OK
Http response payload: {"boolean":true}
Http request: PUT http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv/pc25spath/Room1_Room/Room1_Room.txt?op=create&user.name=pcoello25 HTTP/1.1
Http response status line: HTTP/1.1 307 Temporary Redirect
Http request: PUT http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv/pc25spath/Room1_Room/Room1_Room.txt?op=CREATE&user.name=pcoello25&data=true HTTP/1.1
Http response status line: HTTP/1.1 201 Created
...

[CHECKING PERSISTED ELEMENTS IN storage.cosmos.lab.fiware.org]
$ curl -X GET "http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: xxxxxxxxxxxxxxxxxxxxx"
{"FileStatuses":{"FileStatus":[{"pathSuffix":"pc25serv","type":"DIRECTORY","length":0,"owner":"pcoello25","group":"pcoello25","permission":"755","accessTime":0,"modificationTime":1477042517928,"blockSize":0,"replication":0}]}}

$ curl -X GET "http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: xxxxxxxxxxxxxxxxxxxxx"
{"FileStatuses":{"FileStatus":[{"pathSuffix":"pc25spath","type":"DIRECTORY","length":0,"owner":"pcoello25","group":"pcoello25","permission":"755","accessTime":0,"modificationTime":1477042517928,"blockSize":0,"replication":0}]}}

$ curl -X GET "http://storage.cosmos.lab.fiware.org:14000/webhdfs/v1/user/pcoello25/pc25serv/pc25spath?op=liststatus&user.name=pcoello25" -H "X-Auth-Token: xxxxxxxxxxxxxxxxxxxxx"
{"FileStatuses":{"FileStatus":[{"pathSuffix":"Room1_Room","type":"DIRECTORY","length":0,"owner":"pcoello25","group":"pcoello25","permission":"755","accessTime":0,"modificationTime":1477042518009,"blockSize":0,"replication":0}]}}

```
* Assignee: @frbattid

### CODE REVIEW

File: doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ centos              6                   273a1eca2d3a        2 weeks ago
 
 ##<a name="section3"></a>Using the image
 ###<a name="section3.1"></a>As it is
-The cygnus-ngsi image (either built from the scratch, either downloaded from hub.docker.com) allows running a Cygnus agent in charge of receiving NGSI-like notifications and persisting them into wide variety of storages: MySQL (Running in a  `mysql` host), MongoDB (running in a  `mongo` host), STH (running in `sth` host) and  CKAN (running in `ckan` host).
+The cygnus-ngsi image (either built from the scratch, either downloaded from hub.docker.com) allows running a Cygnus agent in charge of receiving NGSI-like notifications and persisting them into wide variety of storages: MySQL (Running in a  `mysql` host), MongoDB (running in a  `mongo` host), STH (running in `sth` host), CKAN (running in `ckan` host) and HDFS (running in `hdfs` host).
```
Comment (frbattid) [inline doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md]: Thanks to this PR I've seen the host names appearing in this paragraph are not correct. They must be `iot-mysql`, `iot-mongo`, etc.

File: docker/cygnus-ngsi/agent.conf [hunk 2]
```diff
@@ -120,4 +144,8 @@ cygnus-ngsi.channels.sth-channel.transactionCapacity = 100
 
 cygnus-ngsi.channels.ckan-channel.type = com.telefonica.iot.cygnus.channels.CygnusMemoryChannel
 cygnus-ngsi.channels.ckan-channel.capacity = 1000
-cygnus-ngsi.channels.ckan-channel.transactionCapacity = 100
\ No newline at end of file
+cygnus-ngsi.channels.ckan-channel.transactionCapacity = 100
+
+cygnus-ngsi.channels.hdfs-channel.type = com.telefonica.iot.cygnus.channels.CygnusMemoryChannel
+cygnus-ngsi.channels.hdfs-channel.capacity = 1000
+cygnus-ngsi.channels.hdfs-channel.transactionCapacity = 100
```
Comment (frbattid) [inline docker/cygnus-ngsi/agent.conf]: Final white line is required by Github :)

Comment (pcoello25) [inline docker/cygnus-ngsi/agent.conf]: Fixed in 4d54b28

File: doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md (hunk 1 above)
Comment (pcoello25) [inline doc/cygnus-ngsi/installation_and_administration_guide/install_with_docker.md]: Fixed in 4d54b28
Additionally to this fix I've added a mistake in `agent.conf` : It was defined a wrong host (fixed now)

**Pull request merged**